src/components/icon/icon.jsx

```diff
@@ -12,14 +12,17 @@ export default function Icon({ className, name, size, stroke, fill, currentColor
   ].filter(Boolean);
 
   return (
-    <svg className={classes.join(' ')} role="img">
+    <svg id={`${a11yKey}-icon`} aria-label={ariaLabel || name.split('-')[1]} className={classes.join(' ')} aria-labelledby={ariaLabelledBy} role={role} >
```

Comment:
Sometimes, the SVG is purely presentational. Think calendar icon and icon-caret-down. These don't do anything since focusing and activating (e.g. clicking) an input will have the desired action. Perhaps we can make the label optional and avoid adding a bad label through the SVG name?

> 2.1 No ARIA is better than Bad ARIA

https://www.w3.org/TR/wai-aria-practices-1.1/#no_aria_better_bad_aria

Reply:
If the SVG is purely presentational, we will not want to have a ID associated with it as well.

Reply:
While we're here, let's expand the JSX attributes into multi-line attributes for better git-ability.